### ISSUE TITLE
Persistence exception when using dynamic aggregation with @inheritance

### ISSUE THREAD

**Issue by shubhamp-zeotap** [ebean-orm/ebean#1740]
## Actual behavior

@rbygrave Hi Rob,

In continuation of this [issue](https://github.com/ebean-orm/ebean/issues/1737), these are the updated models now I am using

```
@Entity
@Inheritance
@Table(name = "segments")
@DiscriminatorColumn(name = "product_type")
public class BaseSegment extends Model {
}
```

```
@Data
@Entity
@Table(name = "segments")
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@AllArgsConstructor
@NoArgsConstructor
@Inheritance
@DiscriminatorValue("target")
public class PartialSegment extends BaseSegment {

    @Id
    @Column(name = "id")
    private Long id;

    @NotNull
    @Index
    @Column(name = "segment_id_zat")
    private Long segmentIdZat;

    @Index
    @ManyToOne
    @JoinColumn(name = "segment_status_id", referencedColumnName = "status_id", nullable = false)
    private SegmentStatus status;

    @NotNull
    @Column(name = "product_type", nullable = false, columnDefinition = "varchar(255) default 'target'")
    String productType = "target";

    public static final Finder<Long, PartialSegment> find = new Finder<Long, PartialSegment>(PartialSegment.class);
}
```

I want to perform aggregation on this model using this query,
```
        List<PartialSegment> segments = Ebean.find(PartialSegment.class)
                .select("segmentIdZat, min(status)")
                .where().eq("segmentIdZat", 108L).findList();

```

But this is generating ```javax.persistence.PersistenceException: Inheritance type for discriminator value [108] was not found?```

Query executed by Ebean:
```
[debug] io.ebean.SQL - txn[1001] select t0.segment_id_zat, min(t0.segment_status_id) from segments t0 where t0.product_type = 'target'  and t0.segment_id_zat = ? group by t0.segment_id_zat; --bind(108)
```

Stack trace of Exception (complete output of e.printStackTrace()):
```
javax.persistence.PersistenceException: Inheritance type for discriminator value [108] was not found?
	at io.ebeaninternal.server.deploy.InheritInfo.readType(InheritInfo.java:228)
	at io.ebeaninternal.server.deploy.InheritInfo.readType(InheritInfo.java:214)
	at io.ebeaninternal.server.query.SqlTreeNodeBean$LoadInherit.initBeanType(SqlTreeNodeBean.java:230)
	at io.ebeaninternal.server.query.SqlTreeNodeBean$Load.initialise(SqlTreeNodeBean.java:443)
	at io.ebeaninternal.server.query.SqlTreeNodeBean.load(SqlTreeNodeBean.java:459)
	at io.ebeaninternal.server.query.CQuery.readNextBean(CQuery.java:456)
	at io.ebeaninternal.server.query.CQuery.hasNext(CQuery.java:544)
	at io.ebeaninternal.server.query.CQuery.readCollection(CQuery.java:575)
	at io.ebeaninternal.server.query.CQueryEngine.findMany(CQueryEngine.java:394)
	at io.ebeaninternal.server.query.DefaultOrmQueryEngine.findMany(DefaultOrmQueryEngine.java:133)
	at io.ebeaninternal.server.core.OrmQueryRequest.findList(OrmQueryRequest.java:451)
	at io.ebeaninternal.server.core.DefaultServer.findList(DefaultServer.java:1552)
	at io.ebeaninternal.server.core.DefaultServer.findList(DefaultServer.java:1528)
	at io.ebeaninternal.server.querydefn.DefaultOrmQuery.findList(DefaultOrmQuery.java:1526)
	at io.ebeaninternal.server.expression.DefaultExpressionList.findList(DefaultExpressionList.java:439)
	at services.SegmentListingService.listSegments(SegmentListingService.java:32)
	at controllers.SegmentControllerV2.listSegments(SegmentControllerV2.java:65)
	at v2.Routes$$anonfun$routes$1.$anonfun$applyOrElse$4(Routes.scala:96)
	at play.core.routing.HandlerInvokerFactory$$anon$7.resultCall(HandlerInvoker.scala:147)
	at play.core.routing.HandlerInvokerFactory$JavaActionInvokerFactory$$anon$3$$anon$4$$anon$5.invocation(HandlerInvoker.scala:116)
	at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:131)
	at play.mvc.Action.lambda$call$0(Action.java:89)
	at java.util.Optional.map(Optional.java:215)
	at play.mvc.Action.call(Action.java:81)
	at play.http.DefaultActionCreator$1.call(DefaultActionCreator.java:33)
	at play.mvc.Action.call(Action.java:66)
	at actions.AuthorizedAccessAction.call(AuthorizedAccessAction.java:16)
	at actions.AuthenticatedAccessAction.call(AuthenticatedAccessAction.java:27)
	at play.mvc.Action.lambda$call$0(Action.java:89)
	at java.util.Optional.map(Optional.java:215)
	at play.mvc.Action.call(Action.java:81)
	at play.core.j.JavaAction.$anonfun$apply$8(JavaAction.scala:191)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:658)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:79)
	at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:72)
	at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:71)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$KeptPromise$Kept.onComplete(Promise.scala:372)
	at scala.concurrent.impl.Promise$KeptPromise$Kept.onComplete$(Promise.scala:371)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.onComplete(Promise.scala:379)
	at scala.concurrent.impl.Promise.transform(Promise.scala:33)
	at scala.concurrent.impl.Promise.transform$(Promise.scala:31)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.transform(Promise.scala:379)
	at scala.concurrent.Future.map(Future.scala:292)
	at scala.concurrent.Future.map$(Future.scala:292)
	at scala.concurrent.impl.Promise$KeptPromise$Successful.map(Promise.scala:379)
	at scala.concurrent.Future$.apply(Future.scala:658)
	at play.core.j.JavaAction.apply(JavaAction.scala:192)
	at play.api.mvc.Action.$anonfun$apply$2(Action.scala:98)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:92)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:49)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

##  Expected behavior

Ebean seems to be identifying DiscriminatorColumn (product_type) and DiscriminatorValue(target) perfectly fine. Additional where clause on other columns should work fine as well. @rbygrave Is this expected or am I messing up somewhere?

**Ebean Version: 11.40.1**

**Comment by rbygrave** [comment]
I strongly suspect this is a bug - what you are doing looks fine.

**Background**: With inheritance ... when reading a sql row there is an expectation in the internals of Ebean that it should expect a discriminator column to proceed the reading of the other row details.  The value from the discriminator column is used to determine the bean type that will be built/populated from the row / JDBC ResultSet.

In this case, there is no discriminator column in the sql ... I suspect Ebean internally is reading the  `108` value - thinking it is a discriminator value (to determine the bean type) and not being able to map it as a valid descriminator value (because it isn't - it is the `t0.segment_id_zat` value).

**Comment by shubhamp-zeotap** [comment]
Hi Rob,

The generated SQL, in fact, **has** the discriminator column and value already set.
```where t0.product_type = 'target'``` Product Type is my discriminator column. Ebean is adding that to where clause properly.

Ebean is reading the bind values as discriminator even through correct discriminator is present in where clause. I tried explicitly selecting the discriminator column in select clause like this 

```
        List<PartialSegment> segments = Ebean.find(PartialSegment.class)
                .select("segmentIdZat, productType, min(status)")
                .setDisableLazyLoading(true)
                .where().eq("segmentIdZat", 108L).findList();
```

But this also generates the same error.

**Comment by rbygrave** [comment]
The fix for this is https://github.com/ebean-orm/ebean/issues/1744 ... assuming that PartialSegment is a type without any sub-types.

In the case where it has no sub-types then we don't need the discriminator column included in the query (to determine the specific type of PartialSegment or one of it's sub-types).

That is, the fix for this is to change the behaviour of Ebean such that for the case of a query on an inheritance type - if that type has no sub-types then don't try and read the discriminator column (Ebean does need to read it because the type is effectively known).

**Comment by rbygrave** [comment]
Fixed by #1744
